### PR TITLE
[DataPipe] Fix OOM when traverse IterDataPipe due to pickling

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#72209 [DataPipe] Fix OOM when traverse IterDataPipe due to pickling**

This PR would fix https://fburl.com/jtfyksyr

Differential Revision: [D33955933](https://our.internmc.facebook.com/intern/diff/D33955933)

### CODE REVIEW

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33955933).

**Review by NivekT — APPROVED** [review]
LGTM with one comment! Thanks for fixing this.

**Comment by NivekT** [comment]
We should also add this PR to 1.11 if possible.

File: torch/utils/data/graph.py [hunk 1]
```diff
@@ -28,9 +26,8 @@ def stub_pickler(obj):
     def getstate_hook(obj):
         state = {}
         for k, v in obj.__dict__.items():
-            if callable(v) or isinstance(v, PRIMITIVE):
-                continue
-            state[k] = v
+            if isinstance(v, (IterDataPipe, tuple)):
```
[the remark this answers is not in the document]
Comment (ejguan) [reply torch/utils/data/graph.py:29]: Makes sense

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33955933).

**Pull request closed without merging**